controllers/subscription.ts

```diff
@@ -0,0 +1,72 @@
+import { CustomRequest, CustomResponse } from "../types/global";
+const { addOrUpdate } = require("../models/users");
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const nodemailer = require("nodemailer");
+const config = require("config");
+const emailSubscriptionCredentials = config.get("emailSubscriptionCredentials");
+
+export const subscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const { email, phoneNumber } = req.body;
+    const userId = req.userData.id;
+    await addOrUpdate(
+      {
+        phoneNumber,
+        email,
+        isSubscribed: true,
+      },
+      userId
+    );
+    return res.status(201).json({
+      message: "user subscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const unsubscribe = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const userId = req.userData.id;
+    await addOrUpdate(
+      {
+        isSubscribed: false,
+      },
+      userId
+    );
+    return res.status(200).json({
+      message: "user unsubscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const sendEmail = async (req: CustomRequest, res: CustomResponse) => {
+  try { 
+    const transporter = nodemailer.createTransport({
+      host: emailSubscriptionCredentials.host,
+      port:  emailSubscriptionCredentials.port,
+      secure: false,
+      auth: {
+        user: emailSubscriptionCredentials.user,
+        pass: emailSubscriptionCredentials.pass,
+      },
+    });
+
+    const info = await transporter.sendMail({
+      from: `"Real Dev Squad" <${emailSubscriptionCredentials.user}>`,
+      // TODO: after approving this  PR we need to send email to TEJAS sir via this API as a POC.
+      to: "dgandhrav@gmail.com",
```

Comment:
- what does "after approving this pr" mean? are we going to send a test mail to @iamitprakash before the pr is merged to develop?
- who's email is hardcoded here and is it temporary? if so, when will it be removed, and what would the flow look like after that?

Reply:
1. "After approving the PR" we would have just update email through another small  PR from @iamitprakash's email.
2. This is my another email for local testing. 

If you and @iamitprakash want I can now put @iamitprakash's email here, in order to save our time for next PR.

Reply:
@yesyash I have Changed the email after discussing with Tejas sir in last call.